src/editor/ui_menus/tool_change_resources_options_menu.cc

```diff
@@ -88,7 +88,7 @@ EditorToolChangeResourcesOptionsMenu::EditorToolChangeResourcesOptionsMenu(
 	// Add resource buttons
 	resources_box_.add_inf_space();
 	const Widelands::Descriptions& descriptions = parent.egbase().descriptions();
-	for (Widelands::DescriptionIndex i = 0; i < descriptions.nr_resources(); ++i) {
+	for (size_t i = 0; i < descriptions.nr_resources(); ++i) {
```

Comment:
```suggestion
	for (Widelands::DescriptionIndex i = 0; i < static_cast<uint16_t>(descriptions.nr_resources()); ++i) {
```

@hessenfarmer @Noordfrees You want me to do that for the ~15 other instances where I changed DescriptionIndex to size_t

Reply:
I'd suggest changing the definition of `nr_resources` and its siblings to return a `DescriptionIndex` instead of `size_t`. It can be guaranteed to fit in 16 bits.

Reply:
well semantically it should deliver the number of existent description indexes which should be equivalent to the highest index but is not an Index itself. so we might change the return type of the nr_*indexedEntity*() to the same basic type  as DescriptionIndex, as it can't be higher then that, but not the DescriptionIndex as it is not the index but an amount.

Reply:
I started making some changes, they're shown in https://github.com/widelands/widelands/pull/5342/commits/1fffc688352009b2e20a824354f7c3417830685c

What do you think so far?

Reply:
checked the changes in the commit you provided. I think this should be the way to go.